helpers/klusterlerconfig/merge.go

```diff
@@ -0,0 +1,63 @@
+package klusterlerconfig
+
+import (
+	"fmt"
+	"reflect"
+
+	klusterletconfigv1alpha1 "github.com/stolostron/cluster-lifecycle-api/klusterletconfig/v1alpha1"
+)
+
+var klusterletConfigMergeFuncs map[string]func(specs []*klusterletconfigv1alpha1.KlusterletConfigSpec, merged *klusterletconfigv1alpha1.KlusterletConfigSpec) error = map[string]func(specs []*klusterletconfigv1alpha1.KlusterletConfigSpec, merged *klusterletconfigv1alpha1.KlusterletConfigSpec) error{
+	"Registries":                  commonMergeFunc("Registries"),
+	"PullSecret":                  commonMergeFunc("PullSecret"),
+	"NodePlacement":               commonMergeFunc("NodePlacement"),
```

Comment:
can we separate by MergeStruct, MergeStrucPtr and MergeString?

Reply:
Is there any specific benifit of seperating to different type of merge functions? Because commonMerge is expected to handle fields in different kinds.

I would prefer to add new merge functions because of we want a different behivor: for example, for the `Registries` field, maybe we don't want the "pick first non-zero value", instead we want to "merge all list into one". Then we can write another merge function for it.

Reply:
I can rename this common function to be more specific.